target/ppc/int_helper.c

```diff
@@ -1706,6 +1706,41 @@ VEXTRACT(uw, u32)
 VEXTRACT(d, u64)
 #undef VEXTRACT
 
+#define VSTRI(NAME, ELEM, SIZE, LEFT) \
+void helper_##NAME(CPUPPCState *env, ppc_avr_t *t, ppc_avr_t *b,    \
+                   target_ulong rc)                                 \
+{                                                                   \
+    const int array_size = ARRAY_SIZE(b->u##SIZE);                  \
+    bool null_found = false;                                        \
+    int i, idx;                                                     \
+                                                                    \
+    for (i = 0; i < array_size; i++) {                              \
+        idx = LEFT ? i : array_size - i - 1;                        \
+        if (b->Vsr##ELEM(idx)) {                                    \
+            t->Vsr##ELEM(idx) = b->Vsr##ELEM(idx);                  \
+        } else {                                                    \
+            null_found = true;                                      \
+            break;                                                  \
+        }                                                           \
+    }                                                               \
+                                                                    \
+    for (; i < array_size; i++) {                                   \
+        idx = LEFT ? i : array_size - i - 1;                        \
+        t->Vsr##ELEM(idx) = 0;                                      \
+    }                                                               \
+                                                                    \
+    if (rc && null_found) {                                         \
```

Comment:
If !rc, we shouldn't set crf[6] at all

Reply:
Nice catch! Fixed in https://github.com/PPC64/qemu/pull/47/commits/a3758d0cddfc20e12fe4f8761b3a746c6ce4a3d6